pipeline/authn/authenticator_cookie_session.go

```diff
@@ -171,13 +179,16 @@ func forwardRequestToSessionStore(r *http.Request, checkSessionURL string, prese
 		return nil, helper.ErrForbidden.WithReason(err.Error()).WithTrace(err)
 	}
 
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		logger.Tracef("Error reading response from remote: %v", err)
+		return json.RawMessage{}, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to read response from remote: %s", err))
+	}
+
 	if res.StatusCode == 200 {
-		body, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			return json.RawMessage{}, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to fetch cookie session context from remote: %+v", err))
-		}
 		return body, nil
-	} else {
-		return json.RawMessage{}, errors.WithStack(helper.ErrUnauthorized)
 	}
+
+	logger.Tracef("Remote returned non-200 status code '%d' with body: %s", res.StatusCode, body)
```

Comment:
It would be a bit better to use structured logging like `logger.WithField("response_code", res.StatusCode).WithField("body", string(body)).Trace` :)

Reply:
@aeneasr fixed